enhancements/multi-arch/dynamic-imagestream-importmode-setting.md

```diff
@@ -0,0 +1,469 @@
+field with the import mode from the image config status.
+5. openshift-apiserver uses the value of the
+import mode in the observed config's
+ImagePolicyConfig field to set the default import
+mode for any newly created imagestream.
+
+Note that a change to the import mode in the image
+config CR would trigger a redeployment of the
+openshift-apiserver
+
+### API Extensions
+
+- image.config.openshift.io: Add
+  `ImagestreamImportMode` config flag to CRD spec
+  and status.
+
+``` 
+type ImageSpec struct {
+...
+...
+// imagestreamImportMode controls the import
+// mode behaviour of imagestreams. It can be set
+// to `Legacy` or `PreserveOriginal`. The
+// default behaviour is to default to Legacy.
+// If this value is specified, this setting is
+// applied to all imagestreams which do not
+// have the value set.
+// +optional
+ImageStreamImportMode imagev1.ImportModeType
+`json:"imagestreamImportMode,omitempty" }`
+```
+```
+type ImageStatus struct {
+...
+...
+// imagestreamImportMode controls the import
+// mode behaviour of imagestreams. It can be set
+// to `Legacy` or `PreserveOriginal`. The
+// default behaviour is to default to Legacy.
+// If this value is specified, this setting is
+// applied to all new imagestreams which do not
+// have the value set.
+// +optional
+ImagestreamImportMode imagev1.ImportModeType
+`json:"imagestreamImportMode,omitempty"
+}
+```
+- openshiftcontrolplane/v1/: Add
+  `ImagestreamImportMode` config flag to
+  ImagePolicyConfig struct.
+
+```
+type ImagePolicyConfig {
+...
+...
+// imagestreamImportMode controls the import
+mode behaviour of imagestreams. It can be set
+to `Legacy` or `PreserveOriginal`. The
+// default behaviour is to default to Legacy.
+// If this value is specified, this setting is
+// applied to all imagestreams which do not
+// have the value set.
+// +optional
+ImagestreamImportMode imagev1.ImportModeType
+`json:"imagestreamImportMode,omitempty"`
+
+}
+```
+- An example CR for `image.config.openshift.io`
+  would look like below if the import mode is set:
+```
+apiVersion: config.openshift.io/v1
+kind: Image
+metadata:
+  name: cluster
+  ...
+  ...
+spec:
+  imagestreamImportMode: PreserveOriginal
+status:
+   internalRegistryHostname: image-registry.openshift-image-registry.svc:5000
+   imagestreamImportMode: PreserveOriginal
+  ...
+  ...
+
+```
+- The ClusterVersion status field CR would look
+  like below after CVO has inferred the payload
+  type:
+```
+apiVersion: v1
+items:
+- apiVersion: config.openshift.io/v1
+  kind: ClusterVersion
+  spec:
+    channel: candidate-4.15
+    desiredUpdate:
+      force: false
+      version: 4.15.0
+  status:
+    availableUpdates:
+    ...
+    capabilities:
+    ...
+    conditions:
+    ...
+    desired:
+    ...
+    history:
+    ...
+    payloadArchitecture: amd64  # could be arm64/ppc64le/s390x/multi
+    ...
+``` 
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+N/A
```

Comment:
On HyperShift, having the hosted-API-side ClusterVersion driving control-plane side functionality (like OpenShift API server behavior) exposes you to some hosted-admin interference (see [this comment][1], explaining why we jump through hoops to avoid touching hosted-API stuff when setting the upstream update service for HyperShift clusters).  Default `importMode` doesn't seem particularly sensitive, and the registry lives on the hosted compute nodes anyway, so maybe routing through ClusterVersion is fine in this case.  But probably worth explaining that issue, and why we're ok going through ClusterVersion here.

[1]: https://github.com/openshift/cluster-version-operator/pull/1035#issue-2133730335

Reply:
added.